tests/test_cache.c

```diff
@@ -7,9 +7,10 @@ void test_cache_invalidate(TestContext *ctx) {
 	for (int i=0;i<32;i++) {
 		for (int j=0;j<32;j++) {
 
-			// Write the whole buffer through cache,
+			// Read/write the whole buffer through cache,
```

Comment:
Why do we need to also read it?

Reply:
That's a good question. I'm not 100% sure, I haven't studied internal cache workings. What I am sure of is that at some point I needed this to make the test pass. Obviously this is really unstable -- it's using a buffer on the stack and the stack might be already in the cache for previous functions -- but anyway I'm sure it was required.

Now that I think of it, I'll try to move the buffer to global data, so that it'll use separate cache lines and maybe I can reproduce this better